Training_25/Program.cs

```diff
@@ -3,11 +3,49 @@
 // Copyright (c) Metamation India.
 // ------------------------------------------------------------------
 // Program.cs
-// Program on main branch.
+// Program on T01 branch - Assignments Q1: NUMBER CONVERSION GAME
 // ------------------------------------------------------------------------------------------------
+using System.Text;
+using static System.Console;
+using static System.Math;
 namespace Training_25;
+
 internal class Program {
    static void Main (string[] args) {
-      Console.WriteLine ("Hello, World!");
+      WriteLine ("Enter a number");
+      string? input = ReadLine ();
+      int num;
+      while (string.IsNullOrWhiteSpace (input) || !int.TryParse (input, out num)) {
+         WriteLine ("Invalid input. Please enter a  number:");
+         input = ReadLine ();
+      }
+      WriteLine ($"Input: {num}");
+      int n = num;
+      if (n == 0) WriteLine ("HEX: 0");
+      else {
+         bool isNegative = n < 0;
+         n = Abs (n);
+         StringBuilder sb = new ();
+         while (n > 0) {
+            int r = n % 16;
+            n /= 16;
+            if (r < 10) sb.Insert (0, r);
+            else
+               sb.Insert (0, (char)('A' + r - 10));
+         }
+         string hexResult = isNegative ? "-" + sb.ToString () : sb.ToString ();
+         WriteLine ($"HEX: {hexResult}");
+      }
+      int b = Abs (num);
+      StringBuilder bn = new ();
+      if (b == 0) bn.Insert (0, 0);
+      while (b > 0) {
+         int r = b % 2;
+         b /= 2;
+         bn.Insert (0, r);
+      }
+      string binResult = (num < 0 ? "-" : "") + bn.ToString ();
+      WriteLine ($"Binary: {binResult}");
+      ReadLine ();

```

Comment:
```suggestion
      ReadLine ();
```
Since this is just to pause the console, you can use
`Console.WriteLine ("Press any key to exit...");
Console.ReadKey();`

Reply:
Mam, earlier I had this option (Automatically close the console when debugging stops) enabled so the console closes immediately, and I used ReadLine to pause it. But now I have disabled that option so shall I omit this line, or I should include it.

Reply:
@StefyRoseTrumpf A setting should not influence the program behavior. The program should work the same way across different systems irrespective of user settings. So, keep this.

Reply:
@v-priya Thanks for clarifying mam, will do.